packages/wallets/src/components/Base/WalletClipboard/__tests__/WalletClipboard.spec.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { useCopyToClipboard, useHover } from 'usehooks-ts';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import useDevice from '../../../../hooks/useDevice';
+import WalletClipboard from '../WalletClipboard';
+
+jest.mock('usehooks-ts', () => ({
+    useCopyToClipboard: jest.fn(),
+    useHover: jest.fn(),
+}));
+
+jest.mock('../../../../hooks/useDevice', () => ({
+    __esModule: true,
+    default: jest.fn(),
+}));
+
+describe('WalletClipboard', () => {
+    let mockCopy: jest.Mock;
+    const mockUseDevice = useDevice as jest.Mock;
+
+    beforeEach(() => {
+        mockCopy = jest.fn();
+        (useCopyToClipboard as jest.Mock).mockReturnValue([null, mockCopy]);
+        (useHover as jest.Mock).mockReturnValue(false);
+        mockUseDevice.mockReturnValue({ isMobile: false });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render the button with copy icon initially', () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+
+        expect(button).toBeInTheDocument();
+        expect(screen.getByTestId('dt_legacy_copy_icon')).toBeInTheDocument();
+    });
+    it('should render the button with won icon when copied', async () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+
+        expect(screen.getByTestId('dt_legacy_won_icon')).toBeInTheDocument();
+    });
+    it('should call copy function with textCopy when button is clicked', async () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+
+        expect(mockCopy).toHaveBeenCalledWith('Sample text to copy');
+    });
+    it('should show tooltip with "Copy" message when hovered', () => {
+        (useHover as jest.Mock).mockReturnValue(true);
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+
+        expect(screen.getByText('Copy')).toBeInTheDocument();
+    });
+    it('should show tooltip with "Copied!" message when clicked and hovered', async () => {
```

Comment:
Probably it would make sense to add 'describe' here and put a couple of tests inside.
for ex.
```suggestion
    describe('when clicked', () => {
       it( 'calls copy function with textCopy', () => ...)
       
       describe('when hovered', () => 
           it('shows tooltip with "Copied!" message', () => ...)
    })
```